src/cubic-b-spline.cc

```diff
@@ -215,6 +215,10 @@ namespace trajectory
 
     typedef boost::numeric::converter<size_type, double> Double2SizeType;
 
+    if (t == knots_[static_cast<size_t>(imax+1)]) //We are exactly at the end of the spline
+    {
+      return imax;
+    }
```

Comment:
If you confirm that this only concerns the non-uniform case, this should be move inside the dichotomy block.